src/amberscript/parser.h

```diff
@@ -35,6 +35,7 @@ namespace amberscript {
 class Parser : public amber::Parser {
  public:
   Parser();
```

Comment:
Do we need to keep this constructor or can we remove it?

Reply:
We can remove this but it would require passing a nullptr in every parsing unit test. That's why I added a default constructor that would pass that nullptr for you.